lib/Doctrine/ORM/Tools/EntityGenerator.php

```diff
@@ -1380,15 +1432,26 @@ protected function generateEntityStubMethod(ClassMetadataInfo $metadata, $type,
             $methodTypeHint =  '\\' . $typeHint . ' ';
         }
 
+        $methodReturnType = null;
+        if ($this->strictTypes) {
```

Comment:
Sorry for not being specific - meant to add a comment to this line. This seems you are mixing two things.
By strict types, we refer to `declare(strict_types=<value>)`, but absence of this declaration does not imply absence of return types. It's perfectly ok to have return types but no strict declaration (thus having weak types).